### ISSUE TITLE
Support for bundle create, fetch, clone

### ISSUE THREAD

**Issue by benjaminy** [libgit2/libgit2#6824]
I've seen some discussion about `git bundle` in some libgit2 language bindings forums, but not in libgit2 itself. How much effort would it take to support this family of commands?

**Comment by ethomson** [comment]
I'm reasonable sure that a `git bundle` is just a packfile with a list of references prefixed. It should be reasonably straightforward if you were interested in contributing.

**Comment by ethomson** [comment]
Yes - thank you for asking. A number of people have given their permission to relicense their Git contributions into libgit2: https://github.com/libgit2/libgit2/blob/main/git.git-authors

So if those people worked on the `bundle` implementation in Git, then it should be okay. If it was touched by other contributors, we should ask them for permission to use GPLv2 with the linking exception before we use / examine that code.

**Comment by benjaminy** [comment]
Sorry if this is a naive question... Is there an easy way to know who has contributed to a particular feature? Just `git log` the file where it's implemented?
